pkg/chunked/compression_linux.go

```diff
@@ -254,14 +258,20 @@ func readZstdChunkedManifest(blobStream ImageSourceSeekable, tocDigest digest.Di
 		if err != nil {
 			return nil, nil, nil, 0, err
 		}
-		decodedTarSplit, err = decodeAndValidateBlob(tarSplit, tarSplitLengthUncompressed, toc.TarSplitDigest.String())
+		fd, err := unix.Open(tmpDir, unix.O_TMPFILE|unix.O_RDWR|unix.O_CLOEXEC, 0o600)
 		if err != nil {
+			return nil, nil, nil, 0, err
```

Comment:
The error produced by open is just the errno, I suggest you wrap this with something like
`&fs.PathError{Op: "open O_TMPFILE", Path: tmpDir, Err: err}`

Reply:
Thanks, fixed